engine/src/test/scala/cromwell/webservice/CromwellApiServiceSpec.scala

```diff
@@ -502,6 +502,61 @@ class CromwellApiServiceSpec extends AsyncFlatSpec with ScalatestRouteTest with
         }
     }
 
+    behavior of "REST API /query GET endpoint"
+    it should "return labels if specified in query additionalKeys param" in {
+      Get(s"/workflows/$version/query?additionalKeys=labels&id=${CromwellApiServiceSpec.ExistingWorkflowId}") ~>
+        akkaHttpService.workflowRoutes ~>
+        check {
+          status should be(StatusCodes.OK)
+          contentType should be(ContentTypes.`application/json`)
+          val results = responseAs[JsObject].fields("results").convertTo[Seq[JsObject]]
```

Comment:
Your `WorkflowJsonSupport` wil allow you to convert from JSON to scala too, if that makes this test easier